OpenEdXMobile/src/main/java/org/edx/mobile/module/analytics/FirebaseEvent.java

```diff
@@ -0,0 +1,263 @@
+     * @param value The value.
+     */
+    public void putFloat(@NonNull String key, float value) {
+        if (isSkippableKey(key)) return;
+        bundle.putFloat(applyRestrictions(key, PARAM_NAME_MAX_CHARS), value);
+    }
+
+    /**
+     * Utility method to apply Firebase's restrictions on String key and Double value
+     * before putting them in {@link #bundle}.
+     *
+     * @param key   The key.
+     * @param value The value.
+     */
+    public void putDouble(@NonNull String key, double value) {
+        if (isSkippableKey(key)) return;
+        bundle.putDouble(applyRestrictions(key, PARAM_NAME_MAX_CHARS), value);
+    }
+
+    /**
+     * Utility method to apply Firebase's restrictions on String key and Long value
+     * before putting them in {@link #bundle}.
+     *
+     * @param key   The key.
+     * @param value The value.
+     */
+    public void putLong(@NonNull String key, long value) {
+        if (isSkippableKey(key)) return;
+        bundle.putLong(applyRestrictions(key, PARAM_NAME_MAX_CHARS), value);
+    }
+
+    /**
+     * Utility method to apply Firebase's restrictions on String key and Boolean value
+     * before putting them in {@link #bundle}.
+     *
+     * @param key   The key.
+     * @param value The value.
+     */
+    public void putBoolean(@NonNull String key, boolean value) {
+        if (isSkippableKey(key)) return;
+        bundle.putBoolean(applyRestrictions(key, PARAM_NAME_MAX_CHARS), value);
+    }
+
+    /**
+     * Utility method to apply Firebase's restrictions on String key and Double value
+     * before putting them in {@link #bundle}.
+     *
+     * @param map The map containing keys and values.
+     */
+    public void putMap(@NonNull Map<String, String> map) {
+        for (Map.Entry<String, String> entry : map.entrySet()) {
+            final String key = entry.getKey();
+            if (isSkippableKey(key)) continue;
+            bundle.putString(applyRestrictions(key, PARAM_NAME_MAX_CHARS),
+                    applyRestrictions(entry.getValue(), PARAM_VALUE_MAX_CHARS));
+        }
+    }
+
+    /**
+     * Sets category and labels to BI events.
+     *
+     * @param category The category.
+     * @param label    The label.
+     */
+    public void addCategoryToBiEvents(@NonNull String category, @Nullable String label) {
+        putString(Analytics.Keys.CATEGORY, category);
+        if (label != null) {
+            putString(Analytics.Keys.LABEL, label);
+        }
+    }
+
+    /**
+     * Applies Firebase based restrictions on a string by truncating it, if it exceeds the
+     * provided characters limit. Also, replaces the non-acceptable characters with underscore.
+     *
+     * @param string   The string to apply restrictions on.
+     * @param maxChars The max number of characters allowed for the string.
+     * @return The updated string.
+     */
+    @NonNull
+    private String applyRestrictions(@NonNull String string, int maxChars) {
```

Comment:
This restriction only applies to the event name and parameter name. Parameter values can be any string, long, or double.